climada/hazard/wildfire.py

```diff
@@ -0,0 +1,1350 @@
+
+        """
+        centroids = Centroids()
+        centroids.set_raster_from_pnt_bounds((firms['longitude'].min(), \
+            firms['latitude'].min(), firms['longitude'].max(), \
+            firms['latitude'].max()), res=res_data/centr_res_factor)
+        centroids.set_meta_to_lat_lon()
+        centroids.set_area_approx()
+        centroids.set_on_land()
+        centroids.empty_geometry_points()
+
+        return centroids
+
+    def _firms_cons_days(self, firms):
+        """ Compute clusters of consecutive days (temporal clusters).
+        
+            An interruption of days_thresh is necessary to be set in two
+            different temporal clusters.
+
+        Parameters
+        ----------
+            firms : pd.DataFrame
+                 FIRMS data
+
+        Returns
+        -------
+            firms : pd.DataFrame
+
+        """
+        LOGGER.debug('Computing clusters of consecutive days.')
+        firms_iter = firms[firms['iter_ev']][['datenum', 'cons_id', 'event_id']]
+        max_cons_id = firms.cons_id.max() + 1
+        for event_id in np.unique(firms_iter.event_id.values):
+
+            firms_cons = firms_iter[firms_iter.event_id == event_id].reset_index()
+
+            # Order firms dataframe per ascending acq_date order
+            firms_cons = firms_cons.sort_values('datenum')
+            sort_idx = firms_cons.index
+            firms_cons = firms_cons.reset_index()
+            firms_cons = firms_cons.drop(columns=['index'])
+
+            # Check if there is more than 1 day interruption between data
+            firms_cons.at[0, 'cons_id'] = max_cons_id
+            max_cons_id += 1
+            for index in range(1, len(firms_cons)):
+                if abs((firms_cons.at[index, 'datenum'] - firms_cons.at[index-1, 'datenum'])) \
+                >= self.days_thres_firms:
+                    firms_cons.at[index, 'cons_id'] = max_cons_id
+                    max_cons_id += 1
+                else:
+                    firms_cons.at[index, 'cons_id'] = firms_cons.at[(index-1), 'cons_id']
+
+            re_order = np.zeros(len(firms_cons), int)
+            for data, order in zip(firms_cons.cons_id.values, sort_idx):
+                re_order[order] = data
+            firms_iter.cons_id.values[firms_iter.event_id == event_id] = re_order
+
+        firms.cons_id.values[firms['iter_ev'].values] = firms_iter.cons_id.values
+        return firms
+
+    @staticmethod
+    def _firms_clustering(firms, res_data, clus_thres):
+        """ Compute geographic clusters and sort firms with ascending clus_id
+        for each cons_id.
+
+        Parameters
+        ----------
+            firms : pd.DataFrame
+                 FIRMS data
+            res_data : float
+                FIRMS instrument resolution in degrees
+            clus_thres : int
+                Clustering factor which multiplies instrument resolution
+
+        Returns
+        -------
+            firms : pd.DataFrame
```

Comment:
Can you describe what has changed in firms?

Reply:
adjusted